password_security/models/res_users.py

```diff
@@ -76,19 +52,41 @@ def _check_password_rules(self, password):
         if not password:
             return True
         company_id = self.company_id
-        password_regex = ['^']
-        if company_id.password_lower:
-            password_regex.append('(?=.*?[a-z])')
-        if company_id.password_upper:
-            password_regex.append('(?=.*?[A-Z])')
-        if company_id.password_numeric:
-            password_regex.append(r'(?=.*?\d)')
-        if company_id.password_special:
-            password_regex.append(r'(?=.*?\W)')
-        password_regex.append('.{%d,}$' % company_id.password_length)
-        if not re.search(''.join(password_regex), password):
-            raise PassError(self.password_match_message())
-        return True
+        message = []
+        if company_id.password_lower and sum(map(str.islower, password)) < \
```

Comment:
Simplify this condition with `if sum(map(str.islower, password)) < company_id.password_lower`. Empty Integer fields are always `0`.

Also, although summing `True` values is perfectly possible, it's not very straightforward to understand. I'd change the method to `len(filter(str.islower, password))`. This is just a matter of taste, not really a requirement 😉